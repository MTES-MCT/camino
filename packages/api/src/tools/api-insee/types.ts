export interface IApiSirenUniteLegalePeriode
  extends IApiSirenUnionUniteLegalePeriodeEtablissmentUnite {
  dateDebut: Date
  dateFin: Date
}

export interface IApiSirenUnionUniteLegalePeriodeEtablissmentUnite {
  nicSiegeUniteLegale: string | null
  denominationUniteLegale: string | null
  denominationUsuelle1UniteLegale: string | null
  nomUniteLegale: string | null
}

export interface IApiSirenUnionUniteLegaleEtablissmentUnite {
  prenomUsuelUniteLegale: string
  sexeUniteLegale: 'F' | 'M' | null
  sigleUniteLegale: string | null
}

export interface IApiSirenEtablissementUnite
  extends IApiSirenUnionUniteLegalePeriodeEtablissmentUnite,
    IApiSirenUnionUniteLegaleEtablissmentUnite {
  categorieEntreprise: string
  categorieJuridiqueUniteLegale: string
  dateCreationUniteLegale: Date | null
}

export interface IApiSirenEtablissementAdresse {
  codeCedexEtablissement: string | null
  codePaysEtrangerEtablissement: string | null
  numeroVoieEtablissement: string | null
  indiceRepetitionEtablissement: string | null
  typeVoieEtablissement: string | null
  libelleVoieEtablissement: string | null
  codePostalEtablissement: string | null
  libelleCommuneEtablissement: string
  libelleCommuneEtrangerEtablissement: string
}

export interface IApiSirenEtablissement {
  adresseEtablissement: IApiSirenEtablissementAdresse
  siren: string
  uniteLegale: IApiSirenEtablissementUnite
}

export interface IApiSirenUniteLegale
  extends IApiSirenUnionUniteLegaleEtablissmentUnite {
  siren: string
  periodesUniteLegale: IApiSirenUniteLegalePeriode[]
}

export interface IApiSirenQuery {
  fault?: {
    code: number
    description: string
    message: string
  }
  error?: {
    // eslint-disable-next-line camelcase
    error_description: string
  }
  header: {
    debut: number
    nombre: number
    message: 'OK' | string
    statut: number
    total: number
  }
}

export interface IApiSirenQueryTypes extends IApiSirenQuery {
  etablissements?: IApiSirenEtablissement[]
  unitesLegales?: IApiSirenUniteLegale[]
}

export interface IApiSirenQueryToken extends IApiSirenQuery {
  result?: {
    // eslint-disable-next-line camelcase
    access_token: string
  }
}
