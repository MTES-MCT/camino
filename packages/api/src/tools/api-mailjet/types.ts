export enum EmailTemplateId {
  DEMARCHE_CONFIRMATION_DEPOT = 3413770,
  ACTIVITES_NOUVELLES = 3794302,
  ACTIVITES_RELANCE = 3785214,
}

export const EmailAdministration = {
  ONF: 'pole.minier@onf.fr',
  PTMG: 'ptmg@ctguyane.fr',
  DGTM: 'mc.remd.deal-guyane@developpement-durable.gouv.fr',
} as const
