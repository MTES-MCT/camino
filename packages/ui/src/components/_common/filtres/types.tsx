import { DomaineId } from 'camino-common/src/static/domaines'
import { TitreTypeTypeId } from 'camino-common/src/static/titresTypesTypes'
import { capitalize } from 'camino-common/src/strings'
import { FunctionalComponent } from 'vue'

export type Props = {
  element: {
    id: `${TitreTypeTypeId}-${DomaineId}` | TitreTypeTypeId
    nom: string
  }
}

export const FiltresTypes: FunctionalComponent<Props> = (props: Props) => {
  return (
    <span>
      <svg width="24" height="24" class="mr-s mb--xs">
        <rect width="24" height="24" class={`svg-fill-pattern-${props.element.id}`} />
      </svg>
      <span class="h6 bold">{capitalize(props.element.nom)}</span>
    </span>
  )
}
