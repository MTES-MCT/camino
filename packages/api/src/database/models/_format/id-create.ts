import { DemarcheId } from 'camino-common/src/demarche.js'
import { CaminoDate } from 'camino-common/src/date.js'
import { DocumentId, EntrepriseDocumentId, documentIdValidator, entrepriseDocumentIdValidator } from 'camino-common/src/entreprise.js'
import { ActiviteDocumentTypeId, DocumentTypeId } from 'camino-common/src/static/documentsTypes.js'
import { randomBytes } from 'node:crypto'
import { TitreId } from 'camino-common/src/titres.js'
import { EtapeId } from 'camino-common/src/etape.js'
import { UtilisateurId } from 'camino-common/src/roles.js'
import { ActiviteDocumentId, activiteDocumentIdValidator } from 'camino-common/src/activite.js'

export const idGenerate = <T extends string = string>(length = 24): T => randomBytes(length / 2).toString('hex') as T

// TODO 2022-09-13 il faudrait ajouter un préfixe dédié aux démarche à l’id.
// pour pouvoir utiliser « is » de Typescript et controler qu’on ne met pas n’importe quel string dans cette méthode
export const newDemarcheId = (value: string = idGenerate()): DemarcheId => {
  return value as DemarcheId
}

export const newTitreId = (value: string = idGenerate()): TitreId => {
  return value as TitreId
}

export const newEtapeId = (value: string = idGenerate()): EtapeId => {
  return value as EtapeId
}

export const newEnterpriseDocumentId = (date: CaminoDate, documentTypeId: DocumentTypeId): EntrepriseDocumentId => {
  const hash = idGenerate(8)

  return entrepriseDocumentIdValidator.parse(`${date}-${documentTypeId}-${hash}`)
}

export const newActiviteDocumentId = (date: CaminoDate, activiteDocumentTypeId: ActiviteDocumentTypeId): ActiviteDocumentId => {
  const hash = idGenerate(8)

  return activiteDocumentIdValidator.parse(`${date}-${activiteDocumentTypeId}-${hash}`)
}

export const newDocumentId = (date: CaminoDate, documentTypeId: DocumentTypeId): DocumentId => {
  const hash = idGenerate(8)

  return documentIdValidator.parse(`${date}-${documentTypeId}-${hash}`)
}

export const newUtilisateurId = (value: string = idGenerate(6)): UtilisateurId => {
  return value as UtilisateurId
}
