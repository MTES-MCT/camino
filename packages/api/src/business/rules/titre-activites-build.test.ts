import { ITitreActivite, ITitreDemarche } from '../../types.js'

import { titreActivitesBuild } from './titre-activites-build.js'

import { describe, expect, test } from 'vitest'
import { toCaminoDate } from 'camino-common/src/date.js'
import { newDemarcheId, newEtapeId, newTitreId } from '../../database/models/_format/id-create.js'
import { ACTIVITES_TYPES_IDS } from 'camino-common/src/static/activitesTypes.js'
describe("construction des activités d'un titre", () => {
  const aujourdhui = toCaminoDate('2021-01-01')

  test("ne crée pas d'activité pour un titre qui n'a pas de phase de démarches", () => {
    const titreActivites1 = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport trimestriel d'exploitation d'or en Guyane"], [2020], aujourdhui, 'titre-id', 'pxm', undefined)

    expect(titreActivites1.length).toEqual(0)

    const titreActivites2 = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport trimestriel d'exploitation d'or en Guyane"], [2020], aujourdhui, 'titre-id', 'pxm', [])

    expect(titreActivites2.length).toEqual(0)

    const titreActivites3 = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport trimestriel d'exploitation d'or en Guyane"], [2020], aujourdhui, 'titre-id', 'pxm', [
      { id: 'demarche-id' } as unknown as ITitreDemarche,
    ])

    expect(titreActivites3.length).toEqual(0)
  })

  test('ne crée pas une activité si elle existe déjà', () => {
    const res = titreActivitesBuild(
      ACTIVITES_TYPES_IDS["rapport d'exploitation (permis et concessions M)"],
      [2018],
      aujourdhui,
      'titre-id',
      'pxm',
      [{ id: 'demarche-id', phase: {} } as unknown as ITitreDemarche],
      [{ typeId: 'gra', annee: 2018, periodeId: 1 }] as ITitreActivite[]
    )

    expect(res.length).toEqual(0)
  })

  test("ne crée pas une activité si sa date de fin n'a pas eu lieu", () => {
    const res = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport trimestriel d'exploitation d'or en Guyane"], [2021], aujourdhui, 'titre-id', 'pxm', [
      { id: 'demarche-id', phase: {} } as unknown as ITitreDemarche,
    ])

    expect(res.length).toEqual(0)
  })

  test('crée des activités', () => {
    const titreActivitesA = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport d'exploitation (permis et concessions M)"], [2018], aujourdhui, 'titre-id', 'pxm', [
      {
        id: newDemarcheId('demarche-id'),
        titreId: newTitreId('titreId'),
        statutId: 'acc',
        typeId: 'oct',
        demarcheDateDebut: toCaminoDate('2018-01-01'),
        demarcheDateFin: toCaminoDate('2018-12-31'),
        etapes: [
          {
            id: newEtapeId('etape-id'),
            titreDemarcheId: newDemarcheId('demarche-id'),
            date: toCaminoDate('2018-01-01'),
            typeId: 'dpu',
            statutId: 'fai',
            substances: ['auru', 'nacl'],
            surface: null,
            ordre: 1,
            communes: null,
          },
        ],
      },
    ])

    expect(titreActivitesA).toMatchSnapshot()

    const titreActivitesB = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport trimestriel d'exploitation d'or en Guyane"], [2018], aujourdhui, 'titre-id', 'pxm', [
      {
        id: 'demarche-id',
        demarcheDateDebut: '2018-01-01',
        demarcheDateFin: '2018-12-31',
        typeId: 'oct',
      } as ITitreDemarche,
    ])

    expect(titreActivitesB).toMatchSnapshot()
  })

  test("ne crée pas d'activité si le titre n'est pas valide pour la période", () => {
    const titreActivites = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport trimestriel d'exploitation d'or en Guyane"], [2018], aujourdhui, 'titre-id', 'pxm', [
      { id: 'demarche-id', phase: {}, type: {} } as unknown as ITitreDemarche,
    ])

    expect(titreActivites.length).toEqual(0)
  })

  test("ne crée pas d'activités si les sections sont vides", () => {
    const titreActivitesA = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport d'exploitation (permis et concessions M)"], [2018], aujourdhui, 'titre-id', 'pxm', [
      {
        id: 'demarche-id',
        statutId: 'acc',
        typeId: 'oct',
        type: { id: 'oct' },
        phase: { dateDebut: '2018-01-01', dateFin: '2018-12-31' },
        etapes: [
          {
            id: 'etape-id',
            date: '2018-01-01',
            typeId: 'dpu',
            statutId: 'fai',
            substances: [],
          },
        ],
      } as unknown as ITitreDemarche,
    ])

    expect(titreActivitesA).toEqual([])

    const titreActivitesB = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport d'exploitation (permis et concessions M)"], [2018], aujourdhui, 'titre-id', 'pxm', [
      {
        id: 'demarche-id',
        statutId: 'acc',
        typeId: 'oct',
        type: { id: 'oct' },
        phase: { dateDebut: '2018-01-01', dateFin: '2018-12-31' },
        etapes: [
          {
            id: 'etape-id',
            date: '2018-01-01',
            typeId: 'dpu',
            statutId: 'fai',
            substances: null,
          },
        ],
      } as unknown as ITitreDemarche,
    ])

    expect(titreActivitesB).toEqual([])

    const titreActivitesD = titreActivitesBuild(ACTIVITES_TYPES_IDS["rapport d'exploitation (permis et concessions M)"], [2018], aujourdhui, 'titre-id', 'pxm', [
      {
        id: 'demarche-id',
        activiteStatutId: 'acc',
        typeId: 'oct',
        type: {},
        phase: { dateDebut: '2018-01-01', dateFin: '2018-12-31' },
      } as unknown as ITitreDemarche,
    ])

    expect(titreActivitesD).toEqual([])
  })
})
