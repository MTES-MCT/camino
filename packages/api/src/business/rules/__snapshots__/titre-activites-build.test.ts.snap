// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`construction des activités d'un titre > crée des activités 1`] = `
[
  {
    "activiteStatutId": "abs",
    "annee": 2018,
    "date": "2019-01-01",
    "periodeId": 1,
    "sections": [
      {
        "elements": [
          {
            "description": "<b>g (gramme)</b> contenu dans les minerais",
            "id": "auru",
            "nom": "or",
            "type": "number",
            "uniteId": "mgr",
          },
          {
            "description": "<b>x 1000 t (millier de tonnes)</b> extrait par abattage net livré",
            "id": "naca",
            "nom": "sel (chlorure de sodium)",
            "type": "number",
            "uniteId": "mtk",
          },
          {
            "description": "<b>x 1000 t (millier de tonnes)</b> extrait en dissolution par sondage et livré raffiné",
            "id": "nacb",
            "nom": "sel (chlorure de sodium)",
            "type": "number",
            "uniteId": "mtk",
          },
          {
            "description": "<b>x 1000 t (millier de tonnes)</b> extrait en dissolution par sondage et livré en dissolution",
            "id": "nacc",
            "nom": "sel (chlorure de sodium contenu)",
            "type": "number",
            "uniteId": "mtk",
          },
        ],
        "id": "substancesFiscales",
        "nom": "Production annuelle",
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Toute information utile à la compréhension de la production déclarée.",
            "id": "texte",
            "optionnel": true,
            "type": "textarea",
          },
        ],
        "id": "complement",
        "nom": "Informations complémentaires",
      },
    ],
    "titreId": "titre-id",
    "typeId": "gra",
  },
]
`;

exports[`construction des activités d'un titre > crée des activités 2`] = `
[
  {
    "activiteStatutId": "abs",
    "annee": 2018,
    "date": "2018-04-01",
    "periodeId": 1,
    "sections": [
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Masse d’or brut en sortie de mine extrait au cours du trimestre (exemple : masse sous la forme de concentré gravimétrique).",
            "id": "orBrut",
            "nom": "Or brut extrait (g)",
            "type": "number",
          },
          {
            "description": "Masse en gramme de l’ensemble des produits contaminés envoyés en traitement au cours du trimestre.",
            "id": "mercure",
            "nom": "Mercure récupéré (g)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Volume total en litre de carburant détaxé consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantDetaxe",
            "nom": "Carburant détaxé (l)",
            "type": "number",
          },
          {
            "description": "Volume total en litre de carburant conventionnel consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantConventionnel",
            "nom": "Carburant conventionnel (l)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pompes actives au cours du trimestre utilisées sur le chantier (pompe à gravier, pompe de relevage…).",
            "id": "pompes",
            "nom": "Pompes actives",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pelles actives au cours du trimestre utilisées sur le chantier (aménagement, exploitation, réhabilitation).",
            "id": "pelles",
            "nom": "Pelles actives",
            "type": "number",
          },
          {
            "description": "Nombre moyen de salariés sur le chantier au cours du trimestre.",
            "id": "effectifs",
            "nom": "Effectifs",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Montant en euros des investissements consentis au cours du trimestre listés à l’<a href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000021850940&cidTexte=LEGITEXT000006069569\\" target=\\"_blank\\" title=\\"Page de l’article - site externe\\" rel=\\"noopener noreferrer\\">article 318 C de l’annexe II du code général des impôts</a>. Afin de bénéficier des déductions fiscales afférentes, les justificatifs attestant de la réalisation effective des investissements sont susceptibles de vous êtres demandés par l’administration.",
            "id": "environnement",
            "nom": "Dépenses relatives à la protection de l’environnement (euros)",
            "type": "number",
          },
        ],
        "id": "renseignements",
        "nom": undefined,
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "id": "1",
            "nom": "Janvier",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 1,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "2",
            "nom": "Février",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 1,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "3",
            "nom": "Mars",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 1,
            "type": "checkboxes",
          },
        ],
        "id": "travaux",
        "nom": "Statut des travaux",
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Toute information sur les événements marquants du trimestre (accident, incident, arrêt ou suspension d’activité en précisant les raisons, évolution de l’exploitation, difficultés rencontrées, etc.).",
            "id": "texte",
            "optionnel": true,
            "type": "textarea",
          },
        ],
        "id": "complement",
        "nom": "Informations complémentaires",
      },
    ],
    "titreId": "titre-id",
    "typeId": "grp",
  },
  {
    "activiteStatutId": "abs",
    "annee": 2018,
    "date": "2018-07-01",
    "periodeId": 2,
    "sections": [
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Masse d’or brut en sortie de mine extrait au cours du trimestre (exemple : masse sous la forme de concentré gravimétrique).",
            "id": "orBrut",
            "nom": "Or brut extrait (g)",
            "type": "number",
          },
          {
            "description": "Masse en gramme de l’ensemble des produits contaminés envoyés en traitement au cours du trimestre.",
            "id": "mercure",
            "nom": "Mercure récupéré (g)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Volume total en litre de carburant détaxé consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantDetaxe",
            "nom": "Carburant détaxé (l)",
            "type": "number",
          },
          {
            "description": "Volume total en litre de carburant conventionnel consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantConventionnel",
            "nom": "Carburant conventionnel (l)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pompes actives au cours du trimestre utilisées sur le chantier (pompe à gravier, pompe de relevage…).",
            "id": "pompes",
            "nom": "Pompes actives",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pelles actives au cours du trimestre utilisées sur le chantier (aménagement, exploitation, réhabilitation).",
            "id": "pelles",
            "nom": "Pelles actives",
            "type": "number",
          },
          {
            "description": "Nombre moyen de salariés sur le chantier au cours du trimestre.",
            "id": "effectifs",
            "nom": "Effectifs",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Montant en euros des investissements consentis au cours du trimestre listés à l’<a href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000021850940&cidTexte=LEGITEXT000006069569\\" target=\\"_blank\\" title=\\"Page de l’article - site externe\\" rel=\\"noopener noreferrer\\">article 318 C de l’annexe II du code général des impôts</a>. Afin de bénéficier des déductions fiscales afférentes, les justificatifs attestant de la réalisation effective des investissements sont susceptibles de vous êtres demandés par l’administration.",
            "id": "environnement",
            "nom": "Dépenses relatives à la protection de l’environnement (euros)",
            "type": "number",
          },
        ],
        "id": "renseignements",
        "nom": undefined,
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "id": "4",
            "nom": "Avril",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 2,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "5",
            "nom": "Mai",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 2,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "6",
            "nom": "Juin",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 2,
            "type": "checkboxes",
          },
        ],
        "id": "travaux",
        "nom": "Statut des travaux",
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Toute information sur les événements marquants du trimestre (accident, incident, arrêt ou suspension d’activité en précisant les raisons, évolution de l’exploitation, difficultés rencontrées, etc.).",
            "id": "texte",
            "optionnel": true,
            "type": "textarea",
          },
        ],
        "id": "complement",
        "nom": "Informations complémentaires",
      },
    ],
    "titreId": "titre-id",
    "typeId": "grp",
  },
  {
    "activiteStatutId": "abs",
    "annee": 2018,
    "date": "2018-10-01",
    "periodeId": 3,
    "sections": [
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Masse d’or brut en sortie de mine extrait au cours du trimestre (exemple : masse sous la forme de concentré gravimétrique).",
            "id": "orBrut",
            "nom": "Or brut extrait (g)",
            "type": "number",
          },
          {
            "description": "Masse en gramme de l’ensemble des produits contaminés envoyés en traitement au cours du trimestre.",
            "id": "mercure",
            "nom": "Mercure récupéré (g)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Volume total en litre de carburant détaxé consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantDetaxe",
            "nom": "Carburant détaxé (l)",
            "type": "number",
          },
          {
            "description": "Volume total en litre de carburant conventionnel consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantConventionnel",
            "nom": "Carburant conventionnel (l)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pompes actives au cours du trimestre utilisées sur le chantier (pompe à gravier, pompe de relevage…).",
            "id": "pompes",
            "nom": "Pompes actives",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pelles actives au cours du trimestre utilisées sur le chantier (aménagement, exploitation, réhabilitation).",
            "id": "pelles",
            "nom": "Pelles actives",
            "type": "number",
          },
          {
            "description": "Nombre moyen de salariés sur le chantier au cours du trimestre.",
            "id": "effectifs",
            "nom": "Effectifs",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Montant en euros des investissements consentis au cours du trimestre listés à l’<a href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000021850940&cidTexte=LEGITEXT000006069569\\" target=\\"_blank\\" title=\\"Page de l’article - site externe\\" rel=\\"noopener noreferrer\\">article 318 C de l’annexe II du code général des impôts</a>. Afin de bénéficier des déductions fiscales afférentes, les justificatifs attestant de la réalisation effective des investissements sont susceptibles de vous êtres demandés par l’administration.",
            "id": "environnement",
            "nom": "Dépenses relatives à la protection de l’environnement (euros)",
            "type": "number",
          },
        ],
        "id": "renseignements",
        "nom": undefined,
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "id": "7",
            "nom": "Juillet",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 3,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "8",
            "nom": "Août",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 3,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "9",
            "nom": "Septembre",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 3,
            "type": "checkboxes",
          },
        ],
        "id": "travaux",
        "nom": "Statut des travaux",
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Toute information sur les événements marquants du trimestre (accident, incident, arrêt ou suspension d’activité en précisant les raisons, évolution de l’exploitation, difficultés rencontrées, etc.).",
            "id": "texte",
            "optionnel": true,
            "type": "textarea",
          },
        ],
        "id": "complement",
        "nom": "Informations complémentaires",
      },
    ],
    "titreId": "titre-id",
    "typeId": "grp",
  },
  {
    "activiteStatutId": "abs",
    "annee": 2018,
    "date": "2019-01-01",
    "periodeId": 4,
    "sections": [
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Masse d’or brut en sortie de mine extrait au cours du trimestre (exemple : masse sous la forme de concentré gravimétrique).",
            "id": "orBrut",
            "nom": "Or brut extrait (g)",
            "type": "number",
          },
          {
            "description": "Masse en gramme de l’ensemble des produits contaminés envoyés en traitement au cours du trimestre.",
            "id": "mercure",
            "nom": "Mercure récupéré (g)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Volume total en litre de carburant détaxé consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantDetaxe",
            "nom": "Carburant détaxé (l)",
            "type": "number",
          },
          {
            "description": "Volume total en litre de carburant conventionnel consommé au cours du trimestre par les travaux réalisés sur le chantier.",
            "id": "carburantConventionnel",
            "nom": "Carburant conventionnel (l)",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pompes actives au cours du trimestre utilisées sur le chantier (pompe à gravier, pompe de relevage…).",
            "id": "pompes",
            "nom": "Pompes actives",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Nombre moyen de pelles actives au cours du trimestre utilisées sur le chantier (aménagement, exploitation, réhabilitation).",
            "id": "pelles",
            "nom": "Pelles actives",
            "type": "number",
          },
          {
            "description": "Nombre moyen de salariés sur le chantier au cours du trimestre.",
            "id": "effectifs",
            "nom": "Effectifs",
            "type": "number",
          },
          {
            "dateDebut": "2018-01-01",
            "description": "Montant en euros des investissements consentis au cours du trimestre listés à l’<a href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000021850940&cidTexte=LEGITEXT000006069569\\" target=\\"_blank\\" title=\\"Page de l’article - site externe\\" rel=\\"noopener noreferrer\\">article 318 C de l’annexe II du code général des impôts</a>. Afin de bénéficier des déductions fiscales afférentes, les justificatifs attestant de la réalisation effective des investissements sont susceptibles de vous êtres demandés par l’administration.",
            "id": "environnement",
            "nom": "Dépenses relatives à la protection de l’environnement (euros)",
            "type": "number",
          },
        ],
        "id": "renseignements",
        "nom": undefined,
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "id": "10",
            "nom": "Octobre",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 4,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "11",
            "nom": "Novembre",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 4,
            "type": "checkboxes",
          },
          {
            "dateDebut": "2018-01-01",
            "id": "12",
            "nom": "Décembre",
            "options": [
              {
                "id": "nonDebutes",
                "nom": "non débutés",
              },
              {
                "id": "exploitationEnCours",
                "nom": "exploitation en cours",
              },
              {
                "id": "arretTemporaire",
                "nom": "arrêt temporaire",
              },
              {
                "id": "rehabilitation",
                "nom": "réhabilitation",
              },
              {
                "id": "arretDefinitif",
                "nom": "arrêt définitif (après réhabilitation)",
              },
            ],
            "periodeId": 4,
            "type": "checkboxes",
          },
        ],
        "id": "travaux",
        "nom": "Statut des travaux",
      },
      {
        "elements": [
          {
            "dateDebut": "2018-01-01",
            "description": "Toute information sur les événements marquants du trimestre (accident, incident, arrêt ou suspension d’activité en précisant les raisons, évolution de l’exploitation, difficultés rencontrées, etc.).",
            "id": "texte",
            "optionnel": true,
            "type": "textarea",
          },
        ],
        "id": "complement",
        "nom": "Informations complémentaires",
      },
    ],
    "titreId": "titre-id",
    "typeId": "grp",
  },
]
`;
